nemo/collections/asr/models/classification_models.py

```diff
@@ -147,112 +344,119 @@ def transcribe(self, paths2audio_files: List[str], batch_size: int = 4, logprobs
-            collate_fn=collate_func,
-            drop_last=config.get('drop_last', False),
-            shuffle=shuffle,
-            num_workers=config.get('num_workers', 0),
-            pin_memory=config.get('pin_memory', False),
+        output2 = os.path.join(os.path.dirname(output), 'decoder_' + os.path.basename(output))
+        output2_descr = qual_name + ' Decoder exported to ONNX'
+        decoder_onnx = self.decoder.export(
+            output2,
+            None,  # computed by input_example()
+            None,
+            verbose,
+            export_params,
+            do_constant_folding,
+            keep_initializers_as_inputs,
+            onnx_opset_version,
+            try_script,
+            set_eval,
+            check_trace,
+            use_dynamic_axes,
         )
 
-    def setup_training_data(self, train_data_config: Optional[Union[DictConfig, Dict]]):
-        if 'shuffle' not in train_data_config:
-            train_data_config['shuffle'] = True
-        # preserve config
-        self._update_dataset_config(dataset_name='train', config=train_data_config)
+        output_model = attach_onnx_to_onnx(encoder_onnx, decoder_onnx, "EDC")
+        output_descr = qual_name + ' Encoder+Decoder exported to ONNX'
+        onnx.save(output_model, output)
+        return ([output, output1, output2], [output_descr, output1_descr, output2_descr])
 
-        self._train_dl = self._setup_dataloader_from_config(config=train_data_config)
 
-        # Need to set this because if using an IterableDataset, the length of the dataloader is the total number
-        # of samples rather than the number of batches, and this messes up the tqdm progress bar.
-        # So we set the number of steps manually (to the correct number) to fix this.
-        if 'is_tarred' in train_data_config and train_data_config['is_tarred']:
-            # We also need to check if limit_train_batches is already set.
-            # If it's an int, we assume that the user has set it to something sane, i.e. <= # training batches,
-            # and don't change it. Otherwise, adjust batches accordingly if it's a float (including 1.0).
-            if isinstance(self._trainer.limit_train_batches, float):
-                self._trainer.limit_train_batches = int(
-                    self._trainer.limit_train_batches
-                    * ceil((len(self._train_dl.dataset) / self.world_size) / train_data_config['batch_size'])
-                )
+class EncDecClassificationModel(_EncDecBaseModel):
+    """Encoder decoder Classification models."""
 
-    def setup_validation_data(self, val_data_config: Optional[Union[DictConfig, Dict]]):
-        if 'shuffle' not in val_data_config:
-            val_data_config['shuffle'] = False
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
 
-        # preserve config
-        self._update_dataset_config(dataset_name='validation', config=val_data_config)
+        if cfg.is_regression_task:
```

Comment:
Do `if 'is_regression_task' in cfg and cfg.is_regression_task:` so that it doesnt crash if `is_regression_task` doesnt exist in config.

Reply:
Shouldn't it be `if 'is_regression_task' not in cfg or cfg.is_regression_task:`? That way we enforce it to be in config with the right value

Reply:
Looking at the posterior comments below, I guess that you prefer to assume that the absence of the flag `is_regression_task` in the config is equivalent to set it to false. Is that correct?

Reply:
> if 'is_regression_task' not in cfg or cfg.is_regression_task:
Yes this is correct. My bad. 
The assumption is that classification is the default - and therefore does not need the `is_regression_task` flag to be present in the config. 

If it is present, and if it is set to True, raise an error in the constructor of the Classification model.

In the regression models, if the flag doesnt exist or if it does exist but is set to False, raise an error.